client/app/iplb/sslCertificate/order/iplb-ssl-certificate-order.controller.js

```diff
@@ -1,50 +1,102 @@
+            )
+                .then(offers => {
+                    this.offers = offers;
+                    this.sslTypes = _.map(offers, "planCode").map(planCode => planCode.replace(/-/g, "_"));
+
+                    // Add separate free certificate in first position.
+                    this.sslTypes.unshift("free");
+
+                    // Select first by default
+                    this.newSsl.type = this.sslTypes[0];
+                })
+        });
+
         this.organizationTypes = this.IpLoadBalancerConstant.organisationTypes;
+        this.newSsl = {};
+        this.saving = false;
 
-        this.urlJson = JSURL.parse("~(~(planCode~'sslgateway-free~duration~'P1M~configuration~(~(label~'domainname~values~(~'test))~(label~'backendserver~values~(~'1.2.3.4)))~option~(~)~quantity~1~productId~'sslGateway))");
+        this.init();
+    }
 
-        this.newSsl = {
-            type: this.sslTypes[0]
-        };
+    init () {
+        this.paidOffers.load();
     }
 
     order () {
         if (this.form.$invalid) {
             return this.$q.reject();
         }
-        this.saving = true;
 
-        let orderConfig;
-        if (this.newSsl.type === "ev") {
-            orderConfig = this.IpLoadBalancerConstant.sslOrders.comodoEv;
-        } else if (this.newSsl.type === "dv") {
-            orderConfig = this.IpLoadBalancerConstant.sslOrders.comodoDv;
-        } else if (this.newSsl.type === "free") {
-            orderConfig = this.IpLoadBalancerConstant.sslOrders.free;
+        this.planCode = this.newSsl.type.replace(/_/g, "-");
+
+        if (this.planCode === "free") {
+            return this.orderFreeCertificate();
+        }
+
+        const sslOffer = this.offers.find(offer => offer.planCode === this.planCode);
+
+        if (!sslOffer) {
+            return null;
         }
 
-        orderConfig.configuration = {
-            domainname: this.newSsl.displayName ? this.newSsl.displayName : this.newSsl.fqdn,
-            backendserver: this.newSsl.fqdn
-        };
+        const options = Object.assign(_.pick(sslOffer.prices[0], [
+            "duration",
+            "pricingMode"
+        ]), {
+            planCode: this.planCode,
+            quantity: 1
+        });
 
-        if (this.newSsl.type === "ev") {
-            orderConfig.option[0].configuration = this.newSsl.option;
-            orderConfig.option[0].configuration.commonName = this.newSsl.displayName;
-            orderConfig.option[0].configuration.dcv_email = this.newSsl.option.email;
-            orderConfig.option[0].configuration.country = this.newSsl.option.countryName;
+        return this.orderPaidCertificate(options);
+    }
+
+    orderFreeCertificate () {
+        const fqdn = this.newSsl.fqdn.split(",");
+
+        this.saving = true;
+        this.IpLoadBalancerSslCertificateService.orderFreeCertificate(this.$stateParams.serviceName, fqdn)
+            .then(() => this.$state.go("network.iplb.detail.ssl-certificate.home"))
```

Comment:
Here you handle it with a finally but next you handle it with a catch, I don't know if it's wanted or not ?

Reply:
Usually this is done in finally, like this.
Here, a stage changement take no time so it's not a problem to show the submit button again because the page will disappear immediately.
See my next comment ;)